tests/analytics_test.py

```diff
@@ -1,22 +1,45 @@
 import os
 import unittest
 from unittest.mock import patch, MagicMock
-from plugins.analytics.public import censys
+from plugins.analytics.public import censys, shodan
 from core import database_arango
 from censys.search import CensysHosts
 from core.schemas import indicator
 from core.config.config import yeti_config
 from core.schemas.indicator import DiamondModel
 from core.schemas.observable import ObservableType
 from core.schemas import observable
+from parameterized import parameterized
 
 
-class AnalyticsTest(unittest.TestCase):
+class AnalyticsTestBase(unittest.TestCase):
+    def check_observables_and_neighbors(self, query, expected_values):
```

Comment:
```suggestion
    def check_observables_and_neighbors(self, entity, expected_values):
```

This is not specifically a query object is it?

It would also be good to have type annotations here, or at least a docstring on how to use this function

Reply:
Done